packages/base-map/src/util.ts

```diff
@@ -1,9 +1,38 @@
+import { LngLatBoundsLike, MapRef, PaddingOptions } from "react-map-gl";
+
 /**
- * Checks if a parameter is actually a function.
- * @param {*} fn The function to call.
- * @returns fn if fn is a function, or a dummy function.
+ * Computes padding dimensions based on roughly 1/20 of the map's canvas dimensions
+ * (under a 2:1 canvas-to-screen pixel ratio).
+ * @param map The map where the bounds fitting is to occur.
+ * @returns The object with the computed padding dimensions.
  */
-export default function callIfValid(fn) {
-  if (typeof fn === "function") return fn;
-  return () => {};
+export function getFitBoundsPadding(map: MapRef): PaddingOptions {
+  const canvas = map.getCanvas();
+  // @ts-expect-error getPixelRatio not defined in MapRef type.
+  const pixelRatio = map.getPixelRatio();
+  const horizPadding = canvas.width / pixelRatio / 10;
+  const vertPadding = canvas.height / pixelRatio / 10;
+
+  return {
+    bottom: vertPadding,
+    left: horizPadding,
+    right: horizPadding,
+    top: vertPadding
+  };
+}
+
+/**
+ * Helper function used in several packages to fit a map to the given bounds,
+ * using padding based on map canvas size.
+ * @param map The map where the bounds fitting is to occur.
+ * @param bounds The bounds to be fit.
+ */
+export function fitMapBounds(map: MapRef, bounds: LngLatBoundsLike): void {
+  map.fitBounds(bounds, {
+    duration: 500,
```

Comment:
Can we re-add `duration: 300,
       essential: false,
       maxDuration: 600,`?

Reply:
This allows the duration length to flex a little depending on how far the map has to pan.

Reply:
I did not have the impression that these defaults, also set somewhere, were used previously, were they?

Reply:
In the only non-story places where we call `fitBounds` (route overlay, trip overlay, transitive overlay), we have been passing `duration: 500` and have not used the other options mentioned above.

Reply:
It's possible? I'm not sure. 500ms is fine for now.